### PR TITLE
feat(docs/concepts): add details for all concepts

### PR DESCRIPTION
**Pull request by GeorgeMac**
This adds details to the concepts pages for the following topics:

- [x] Resource Definitions
- [x] Controllers
- [x] Bindings

### CODE REVIEW

File: docs/site/pages/concepts/controllers.mdx [hunk 1]
```diff
@@ -8,4 +8,245 @@ Head to the configuration reference section for [Controllers](/configuration/con
...
+Controllers declare _how_ to handle storing, retrieving and removing resource kinds from a repository.
+
+Requests to Cup for a particular resource (get, list, put or delete) ultimately end up being handled by a **controller**.
+By default, Flipt ships with a basic controller known as the `template` controller and an extensible controller called the `wasm` controller.
+
+<Callout type="info" emoji="ℹ️">
+You will learn in the next section on [bindings](/concepts/bindings) how to configure which controllers handle which resource definitions.
+</Callout>
+
+Controllers are the most important extension point in Flipt.
```
Comment (yquansah) [inline docs/site/pages/concepts/controllers.mdx:20]:
```suggestion
Controllers are the most important extension point in Cup.
```

Did this mean to say `cup` or `Flipt`?

File: docs/site/pages/concepts/controllers.mdx [hunk 2]
```diff
@@ -8,4 +8,245 @@ Head to the configuration reference section for [Controllers](/configuration/con
...
+    G -->>- C: 
+    C -->>- S: return
+    S ->> SCM: OpenPR(branch, title)
+    SCM -->> S: PR{}
+    S -->>- A: 202 Accepted
+```
+
+Write requests on the other hand are a little more complicated.
+In this scenario, the filesystem abstraction passed to a Controller can handle writes.
+The API server which builds the implementation of the filesystem, will do so over a worktree on a new branch.
```
Comment (yquansah) [inline docs/site/pages/concepts/controllers.mdx:85]:
```suggestion
In Git's case, the API server which builds the implementation of the filesystem, will do so over a worktree on a new branch.
```
You can take it or leave it.

I had to piece it a bit together. Are there any other FS abstractions that cup supports?

File: docs/site/pages/concepts/controllers.mdx [hunk 3]
```diff
@@ -8,4 +8,245 @@ Head to the configuration reference section for [Controllers](/configuration/con
...
+        └──────────────────────┘                               
+```
+
+The purpose of this subcommand is to remove an existing resource.
+Implementations should adjust the filesystem appropriately for the resource type and controllers needs.
+The namespace and name of the resource is passed as arguments to the subcommand.
+
+TBD:
+
+- What makes sense to return from the binary?
```
Comment (yquansah) [inline docs/site/pages/concepts/controllers.mdx:252]: We have a sample `wasm` controller written for Flipt resource definitions right? I wonder if it makes sense to link to that somewhere here for reference.

**Review by yquansah — COMMENTED** [review]
Looks amazing overall.

File: docs/site/pages/concepts/controllers.mdx (hunk 1 above)
Comment (GeorgeMac) [reply docs/site/pages/concepts/controllers.mdx:20]: I do this all the time 🤦 😂 Thank you!

File: docs/site/pages/concepts/controllers.mdx (hunk 2 above)
Comment (GeorgeMac) [reply docs/site/pages/concepts/controllers.mdx:85]: Yeah it supports local too. Like with Flipt.

File: docs/site/pages/concepts/controllers.mdx (hunk 3 above)
Comment (GeorgeMac) [reply docs/site/pages/concepts/controllers.mdx:252]: That sounds like a great idea.

Comment (GeorgeMac) [reply docs/site/pages/concepts/controllers.mdx:256]: I added a callout near the top of the WASM section. Let me know what you think 👍

File: docs/site/pages/concepts/controllers.mdx [hunk 4]
```diff
@@ -8,4 +8,249 @@ Head to the configuration reference section for [Controllers](/configuration/con
...
+The listing template produces a glob syntax path. This glob path will be used on `list` operations to locate all the files containing definitions for resources of a particular kind in a particular namespace.
+The resource template identifies a single path for a single named resource.
+
+Both of these templates can be overriden via Cups controller configuration.
+Head to [Configuration: Controller: Template](/configuration/controllers#template) to learn how.
+
+### WASM Controller
+
+The WASM controller leverages the [Wazero](https://wazero.io/) runtime for Go to support implementing Controllers in the language of your choice.
+WASM binary implementations should expose a command-line interfaces with a number of subcommands.
```
Comment (markphelps) [inline docs/site/pages/concepts/controllers.mdx:130]:
```suggestion
WASM binary implementations should expose a command-line interface with a number of subcommands.
```

File: docs/site/pages/concepts/definitions.mdx [hunk 5]
```diff
@@ -9,3 +9,48 @@ to learn how they can be configured.
 </Callout>
 
 Resource definitions describe the schema for the different resource kinds available in a Cup instance.
+When you interact with the Cup API, you do so through the various resource definition configured.
```
Comment (markphelps) [inline docs/site/pages/concepts/definitions.mdx:12]:
```suggestion
When you interact with the Cup API, you do so through the various configured resource definitions.
```

**Review by markphelps — APPROVED** [review]
couple minor wording changes, otherwise lgtm

**Pull request merged**